src/Resolution.ts

```diff
@@ -94,18 +101,97 @@ export default class Resolution {
 
   /**
    * Creates a resolution instance with configured provider
-   * @param provider - any provider with sendAsync function impelmented
+   * @param provider - any provider compatable with EIP-1193 (https://eips.ethereum.org/EIPS/eip-1193)
    */
-  static provider(provider: Web3Provider): Resolution {
-    return new this({ blockchain: { web3Provider: provider } });
+  static provider(provider: Provider): Resolution {
+    return new this({ blockchain: { provider: provider } });
   }
 
   /**
    * Creates a resolution instance from configured jsonRPCProvider
    * @param provider - any jsonRPCprovider will work as long as it's prototype has send(method, params): Promise<any> method
    */
   static jsonRPCprovider(provider): Resolution {
```

Comment:
I guess it was never working as designed well, so I'd also take opportunity to rename it to `fromEthersJsonRpcProvider`